src/etc/inc/dyndns.class

```diff
@@ -1117,7 +1122,92 @@
+						return false;
+					} else if ( $http_code == 401 ) {
+						log_error("$err_prefix " . gettext("querying domains") . ": " . gettext("User Authorization Failed"));
+						return false;
+					} else if ( $http_code != 200 ) {
+						log_error("$err_prefix " . gettext("querying domains") . ": " .
+							( isset($domains_result["errors"][0]["reason"]) ? $domains_result["errors"][0]["reason"] : "HTTP $http_code" ) );
+						return false;
+					}
+
+					$domains_result = json_decode($domains_output, TRUE);
+					foreach($domains_result["data"] as $domains_entry) {
+						if ($domains_entry["domain"] == $this->_dnsDomain) {
+							$domain_id = $domains_entry["id"];
+						}
+					}
+					if ( ! $domain_id ) {
+						log_error("$err_prefix " . gettext("no domain ID for domain") . ": '{$this->_dnsDomain}'");
+						return false;
+					}
+					if ($this->_dnsVerboseLog) {
+						log_error("_update(): " . sprintf(gettext("found domain id: %s"), $domain_id));
+					}
+
+					// get existing record if present
+					curl_setopt($ch, CURLOPT_URL, "$linode_api/domains/$domain_id/records");
+					$records_output = curl_exec($ch);
+					$http_code = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+					if ( $http_code != 200 )
+					{
+						log_error("$err_prefix " . gettext("querying domains") . ": " .
+							( isset($domains_result["errors"][0]["reason"]) ? $domains_result["errors"][0]["reason"] : "HTTP $http_code" ) );
+						return false;
+					}
+
+					$records_result = json_decode($records_output, TRUE);
+					foreach($records_result["data"] as $records_entry) {
+						if ( $records_entry["type"] == $record_type && $records_entry["name"] == $record_name ) {
+							// not adding support for pagination at this time, hope you have < 100 records!
+							$record = $records_entry;
+						}
+					}
+					if ($this->_dnsVerboseLog) {
+						log_error("_update(): " . ( $record ? sprintf(gettext("found existing record id: %s"), $record["id"]) : gettext("no matching record found") ));
+					}
+
+					if (is_array($record)) {
+						// update existing record
+						$record["target"] = $this->_dnsIP;
+						$record["ttl_sec"] = (int) $this->_dnsTTL; // linode may round this up, 0 = zone default
+						curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($record));
+						curl_setopt($ch, CURLOPT_CUSTOMREQUEST, 'PUT');
+						curl_setopt($ch, CURLOPT_URL, "$linode_api/domains/$domain_id/records/" . $record["id"]);
+					} else {
+						// create a new record
+						$record = array(
+							"type"    => $record_type,
+							"name"    => $record_name,
+							"target"  => $this->_dnsIP,
+							"ttl_sec" => (int) $this->_dnsTTL, // linode may round this up, 0 = zone default
+						);
+						curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($record));
+						curl_setopt($ch, CURLOPT_CUSTOMREQUEST, 'POST');
+						curl_setopt($ch, CURLOPT_URL, "$linode_api/domains/$domain_id/records");
```

Comment:
Variable references in strings should have braces around them, e.g. `{$linode_api}` ( https://www.netgate.com/docs/pfsense/development/developer-style-guide.html#general-rules )